src/common/Preforker.h

```diff
@@ -45,6 +45,8 @@ class Preforker {
       return (errno = e, -1);
     }
 
+    signal(SIGHUP, SIG_IGN);
```

Comment:
instead of using `signal()`, i'd suggest use `sigaction()`, like
```suggestion
    sigaction sa;
    sigemptyset(&sa.sa_mask);
    sa.sa_flags = 0;
    sa.sa_handler = SIG_IGN;
    if (sigaction(SIGHUP, &sa, nullptr) == -1) {
      int e = errno;
      oss << "[" << getpid() << "]: unable to ignore SIGHUP: " << cpp_strerror(e);
      err = oss.str();
      return (errno = e, -1);
    }
```

Reply:
Thank you for your advice，I have modified it.